zulipterminal/ui_tools/views.py

```diff
@@ -1144,7 +1156,22 @@ def _fetch_user_data(controller: Any, user_id: int) -> Dict[str, Any]:
             if data["last_active"]:
                 display_data["Last active"] = data["last_active"]
 
-        return display_data
+        # This will be an empty dict in case of bot users
+        custom_profile_data = data["custom_profile_data"]
+
+        if custom_profile_data:
+            for field in custom_profile_data:
+                # Get label, type, order and value of field data
+                if field["type"] == 6:  # Person picker
+                    user_names = [
+                        controller.model.user_name_from_id(user)
+                        for user in field["value"]
+                    ]
+                    field["value"] = ", ".join(user_names)
+
+                display_custom_profile_data[field["label"]] = str(field["value"])
```

Comment:
It's only purpose is to indicate to mypy that field["value"] is a string instead of being a `Union[str, List[int]]` since `display_custom_profile_data` is typed as being  a `Dict[str,str]` (part of return value of the function).
I could change the type to `Dict[str, Any]` which would allow `str` to be removed but that might make the typing a bit weaker.

Reply:
Ah, I see the issue here. Generally if I see an explicit type conversion in code like this then I'm expecting there's a runtime reason, like the conversion from an int for lookup in the string-int index of custom profile fields.

Using `str()` is totally valid, and so mypy will also accept that as a 'fix' in this case.

However, for this kind of situation it's useful to use either a `cast` (or similar) as an indication to mypy and the reader what is happening and add a comment, or fix the type issue ;) We don't need an actual runtime conversion, as it's a string already, and a cast avoids that and indicates "I know what I'm doing - see the comment" ;)

Fixing the typing fully is likely too much work here, since we'd have to map from literals to the type of the data.

Another approach which is used a lot instead of `cast` is to use an assert on the type. For example, here you're handling the only non-string case specially, so an `else` should be fine:
```python3
for field in custom_profile_data:
    if field["type"] == 6:  # Person picker
        ...
    else:
        assert isinstance(field["value"], str)
```
Or, since you're reassigning `field["value"]`, you can likely move the assert outside of the conditional.
A comment may still be useful though :)
This is (typically) a runtime check, but doesn't implicitly convert anything like `str` would.